LICENSE.txt

```diff
@@ -6,34 +6,39 @@ Most of the Checker Framework is licensed under the GNU General Public
 License, version 2 (GPL2), with the classpath exception.  The text of this
 license appears below.  This is the same license used for OpenJDK.
 
-A few parts of the Checker Framework have more permissive licenses.
+A few parts of the Checker Framework have more permissive licenses, notably
+the parts that you might want to include with your own program.
 
  * The annotations and utility files are licensed under the MIT License.
-   (The text of this license also appears below.)  More specifically, all
-   the parts of the Checker Framework that you might want to include with
-   your own program use the MIT License.  This is the checker-qual*.jar and
-   checker-compat-qual*.jar files and all the files that appear in them:
-   every file in a qual/ directory, plus utility files such as
-   NullnessUtil.java, RegexUtil.java, SignednessUtil.java, etc.  In
-   addition, the cleanroom implementations of third-party annotations,
-   which the Checker Framework recognizes as aliases for its own
-   annotations, are licensed under the MIT License.
-
-Some external libraries that are included with the Checker Framework have
-different licenses.
+   (The text of this license also appears below.)  This applies to the
+   checker-qual*.jar and all the files that appear in it:  every file in a
+   qual/ directory, plus utility files including FormatUtil.java,
+   I18nFormatUtil.java, NullnessUtil.java, Opt.java,
+   PurityUnqualified.java, RegexUtil.java, SignednessUtil.java, and
+   UnitsTools.java.  It also applies to the cleanroom implementations of
+   third-party annotations (e.g., in checker/src/testannotations/ and in
+   framework/src/main/java/org/jmlspecs/).
+
+The Checker Framework includes annotations for some libraries (e.g., in
+.astub files and in https://github.com/typetools/jdk, which appears in the
+annotated-jdk directory of file checker.jar).  Each annotated library uses
+the same license as the original upstream version of the library.
```

Comment:
This is the only problematic part. I count 55 .astub files in this repository. None of them (of the random sampling I looked at) contains any license information. If they are under a different license than the main source code then that needs to be explicitly stated to allow legally-safe use of this library. Putting a LICENSE file in each directory would be fairly arduous but a summary in this file should suffice. e.g. 
```suggestion
annotated-jdk directory of file checker.jar).  Each annotated library uses
the same license as the original upstream version of the library.
Licenses for .astub files are summarized below:
checker/jtreg/*            license A
framework/tests/report/*   license B
```